File: 2019-2020/pset6/mario.py
# https://docs.cs50.net/2019/x/psets/6/sentimental/mario/more/mario.html
# https://cs50.harvard.edu/x/2020/psets/6/mario/more/

# Mario, more comfortable
# Implement a program that prints out a double half-pyramid of a specified
# height, per the below.

from cs50 import get_int

while True:
    height = get_int("Height: ")
    if 0 < height < 9:
        break

for k in range(height):
    print('{0}{1}  {1}'.format((height - k - 1) * ' ', (k + 1) * '#'))

File: 2014-2016/pset1/hacker/mario.c
// Hacker's version
// http://docs.cs50.net/problems/mario/more/mario.html
// https://cdn.cs50.net/2014/fall/psets/1/hacker1/hacker1.html#itsa_mario

#include <stdio.h>
#include "cs50.h"   

void print_chars(char c, int n) {
    for (int i = 0; i < n; ++i) {
        printf("%c", c);
    }
}

int main(void)
{

    char sharp = '#';
    char blank = ' ';
    int n;

    do {
        printf("Height: ");
        n = GetInt();
        if (n == 0)
            return 0;
    } while ((n < 1) || (n > 23));


    for (int i = 1; i <= n; ++i) {
        print_chars(blank, n - i);
        print_chars(sharp, i);
        print_chars(blank, 2);
        print_chars(sharp, i);
        printf("\n");        
    }
 
    return 0;
}